src/fdc3/dotnet/DesktopAgent.Client/src/MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client/Infrastructure/DesktopAgentClient.cs

```diff
@@ -150,14 +151,29 @@ public async Task<IImplementationMetadata> GetInfo()
         return implementationMetadata;
     }
 
-    public Task<IChannel> GetOrCreateChannel(string channelId)
+    public async Task<IChannel> GetOrCreateChannel(string channelId)
     {
-        throw new NotImplementedException();
+        if (_appChannels.TryGetValue(channelId, out var existingChannel))
```

Comment:
This should be a GetOrCreate with a factory method on the ConcurrentDictionary instead to prevent race conditions

Reply:
We should have to check if the app channel is already captured because if not then we should send a message to the backend and if it is captured then we should just return that channel.